File: 2294_Partition_Array_Such_That_Maximum_Difference_Is_K/solution.py
class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
        nums.sort()
        m, count = nums[0], 1
        for i in range(len(nums)):
            if nums[i] - m > k:
                count+=1
                m = nums[i]
        return count

File: 2294_Partition_Array_Such_That_Maximum_Difference_Is_K/solution.c
int comp(const void* a, const void* b) { return (*(int*)a - *(int*)b); }
int partitionArray(int* nums, int numsSize, int k) {
    qsort(nums, numsSize, sizeof(nums[0]), comp);
    int min = nums[0], count = 1;
    for (int i = 0; i < numsSize; i++) if (nums[i] - min > k) { count++, min = nums[i]; }
    return count;
}